### ISSUE TITLE
Plugin doesn't fetch/respect the timezone of task due dates

### ISSUE THREAD

**Issue by pmason314** [jamiebrynes7/obsidian-todoist-plugin#371]
**Describe the bug**
Hi, I really appreciate this plugin and the 2.0.0 changes, but believe it introduced a timezone-related bug.  The plugin correctly fetches my tasks and their due dates, but it consistently lists the due date time as 8 hours early (which is coincidentally the difference between UTC and my timezone of PST).

**To Reproduce**
1. Create a task in Todoist with a designated due date time, say 12pm on Sunday.
2. In Todoist's Settings -> General, set the timezone to US/Pacific (any non-UTC option will do).
3. Sync/refresh the Todoist plugin query in Obsidian (the query I used is below).
4. Observe that the rendered block in Obsidian from the query shows a due time of 4am Sunday instead of 12pm Sunday.

**Expected behavior**
The task's due time in Obsidian should match what is shown in Todoist (which I believe was the case pre-2.0.0).  Ideally this would happen from retrieving the timezone from the connected Todoist account, but if that's possible then a timezone option in the Obsidian plugin settings would also work.

**Screenshots**
None applicable

**Desktop (please complete the following information):**
 - Plugin Version: 2.0.0
 - Obsidian Version 1.7.7

**Additional context**
The query I used to produce the mismatched timezone:
```todoist
name: Upcoming Tasks
filter: "(today | overdue | 7 days | no due date)"
sorting:
  - date
show:
  - due
```

Thank you!

**Comment by jamiebrynes7** [comment]
Hey there, the plugin is intended to display tasks in your system's current timezone, this was the case before the 2.0.0 release as well. 

Can you open the dev tools in Obsidian and run the following in the console:

```
new Intl.DateTimeFormat().resolvedOptions().timeZone
```

This is what the plugin (eventually) relies on for the timezone and at least, for me, this is correct.

**Comment by jamiebrynes7** [comment]
I can actually see that the API response from Todoist doesn't match what their API docs say it should 😞 

Previously I think the plugin would parse the dates as the local timezone and it would just work, provided you were in the same timezone as your Todoist account. Now we try and handle the response as the API suggests and it results in the discrepancy you see. I live in UTC, so I didn't notice this when I made the changes!

I can patch this so we go back to the previous behaviour (same timezone works, different doesn't).

**Comment by pmason314** [comment]
Thanks for the quick response!  Confirming that `new Intl.DateTimeFormat().resolvedOptions().timeZone` gives me `'America/Los_Angeles'`, but gotta love out of date/inaccurate API docs 🙃 .  The previous behavior would be great - thanks again!

**Comment by jamiebrynes7** [comment]
Turns out its slightly more complex than I originally thought! Todoist has [floating times](https://todoist.com/help/articles/set-a-fixed-time-or-floating-time-for-a-task-YUYVp27q), meaning by default the time will stay the same even if your timezone changes. Only if specifically set, will the task have a timezone associated with it.

This is reflected in the API as well, so when there is a specific timezone we get a datetime with a trailing `Z` indicating its fixed - we can just check for that and handle it accordingly. I think this may have worked by accident before 😆

**Comment by jamiebrynes7** [comment]
Hey @pmason314, just released v2.0.1 which should fix this!

**Comment by pmason314** [comment]
Amazing!  Confirming that I'm seeing the correct due time now - have a good one!
